pennylane_qiskit/qiskit_device2.py

```diff
@@ -550,21 +558,33 @@ def _execute_runtime_service(self, circuits, session):
             else self.backend.configuration().backend_name
         )
 
-        options = {
+        circuit_runner_options = {
             "backend": backend_name,
             "log_level": self.options.environment.log_level,
             "job_tags": self.options.environment.job_tags,
             "max_execution_time": self.options.max_execution_time,
         }
 
         # Send circuits to the cloud for execution by the circuit-runner program.
-        job = self.service.run(
-            program_id="circuit-runner",
-            options=options,
-            inputs=program_inputs,
-            session_id=session.session_id,
-        )
-        self._current_job = job.result(decoder=RunnerResult)
+        # Cloud simulators will be deprecated on May 15th so this will be exclusively for real hardware devices.
+        if self.service:
+            job = self.service.run(
+                program_id="circuit-runner",
+                options=circuit_runner_options,
+                inputs=program_inputs,
+                session_id=session.session_id,
+            )
+            self._current_job = job.result(decoder=RunnerResult)
+        else:  # Uses local simulator instead. After May 15th, all simulations will use this logic instead.
+            # Does not support AerSimulator specific options e.g. choose a specific method
+            # refer to this https://qiskit.github.io/qiskit-aer/stubs/qiskit_aer.AerSimulator.html
+            # To support this would be confusing in terms of option setting (The options we currently support are runtime options)
+            # This here is just to capture and track shots information
+            self.backend.set_options(shots=shots)
+            job = self.backend.run(
```

Comment:
Do we have tests for executing this, or is writing them not so straightforward?

Reply:
The tests for tracking the shots for the local simulator exist and are passing. It's not clearly reflected in the PR but the test `test_tape_shots_used_runtime_service` has had its backend updated to use the AerSimulator instead of the cloud simulator and the test is passing.